### PR TITLE
PUBDEV-5894 / PUBDEV-5820 - CDH 6.0

### PR DESCRIPTION
**Pull request by Pscheidl**
This branch contains commits from https://github.com/h2oai/h2o-3/pull/3092, as this PR is not merged yet and we wanted to test it.

### CODE REVIEW

**Comment by michalkurka** [comment]
@Pscheidl I changed the base of this PR to `pavel_pubdev_5820_jetty_9` to see just the changes this PR introduces

File: h2o-hadoop/h2o-cdh6.0/CDH60.Makefile [hunk 1]
```diff
@@ -0,0 +1,100 @@
+CDHV=6.0
```
Comment (michalkurka) [inline h2o-hadoop/h2o-cdh6.0/CDH60.Makefile]: does this file belong in this directory? no other hadoop distributions have it

cc: @michal-raska

File: h2o-hadoop/h2o-cdh6.0/CDH60.Makefile [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+clean:
+	./gradlew clean
+
+build: .PHONY
+	./gradlew build -x test
+
+run:
+	java \
+		-Djetty.request.header.size=32768 \
```
Comment (michalkurka) [inline h2o-hadoop/h2o-cdh6.0/CDH60.Makefile]: why is this necessary?

File: h2o-hadoop/h2o-cdh6.0/CDH60.Makefile (hunk 1 above)
Comment (michal-raska) [reply h2o-hadoop/h2o-cdh6.0/CDH60.Makefile]: No, I think this is a "helper", will remove.

File: h2o-hadoop/h2o-cdh6.0/CDH60.Makefile (hunk 2 above)
Comment (Pscheidl) [reply h2o-hadoop/h2o-cdh6.0/CDH60.Makefile]: Whole file is a leftover, will be deleted. I copy&pasted past changes and made changes on my own in the past commits. Forgot about this file.

File: h2o-parsers/h2o-parquet-compat/h2o-parquet-v17-compat/build.gradle [hunk 3]
```diff
@@ -22,6 +22,7 @@ dependencies {
 
   testCompile "junit:junit:${junitVersion}"
   testCompile project(path: ":h2o-core", configuration: "testArchives")
+  testCompile project(":h2o-jetty-${defaultJettyVersion}")
```
Comment (pkozelka) [inline h2o-parsers/h2o-parquet-compat/h2o-parquet-v17-compat/build.gradle]: @michal-raska I think this dependency should be `testRuntimeOnly` as we probably don't need to compile tests against it.

... but anyway, do we really need it? It is already packaged into runtime jar by `test-package-java` so this should not be necessary at all

Comment (jakubhava) [reply h2o-parsers/h2o-parquet-compat/h2o-parquet-v17-compat/build.gradle]: Please see https://github.com/h2oai/h2o-3/pull/3115#issuecomment-444778813

- We should not use specific versions outside of fat jars
- I think we decided to remove `defaultJettyVersion` to avoid confusion

**Comment by Pscheidl** [comment]
Resolved the generic module dependencies today. We'll need to introduce a new mapreduce-generic module. Easy task if we know how the dependencies should look like:

```groovy
ext {
  hadoopVersion = 'cdh6.0.1'
  hadoopMavenArtifactVersion = '3.0.0-cdh6.0.1'
}

apply plugin: 'java'

description = 'H2O mapreduce generic driver'

sourceCompatibility = 1.7
targetCompatibility = 1.7

compileJava { 
  options.debug = true 
}

dependencies {
  compile('org.apache.hadoop:hadoop-client:' + hadoopMavenArtifactVersion)
  compile project(':h2o-app')
  compile project(":h2o-web")
  compile project(":h2o-avro-parser")
  compile project(":h2o-webserver-iface")
  compile 'org.apache.hadoop:hadoop-mapreduce-client-app:3.0.0-cdh6.0.1'
  compile 'org.apache.hive:hive-jdbc:2.1.1-cdh6.0.1'
  compile 'org.apache.hive:hive-common:2.1.1-cdh6.0.1'
}

```


Also, this issue must be resolved:

```plain
12-07 12:22:39.984 10.20.30.189:54445    2256   #44334-31 INFO: ExportFiles processing (s3n://test.0xdata.com/h2o-hadoop-tests/test-export/s3n/exported.20181207-122239.a47dd5dd-ad69-4177-84d6-8be891d9b4ce.csv.zip)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: water.api.HDFSIOException: HDFS IO Failure: 
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR:  accessed URI : s3n://test.0xdata.com/h2o-hadoop-tests/test-export/s3n/exported.20181207-122239.a47dd5dd-ad69-4177-84d6-8be891d9b4ce.csv.zip
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR:  configuration: Configuration: core-default.xml, core-site.xml, mapred-default.xml, mapred-site.xml, yarn-default.xml, yarn-site.xml, hdfs-default.xml, hdfs-site.xml, /etc/hadoop/conf/core-site.xml
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR:  java.io.IOException: The s3n:// client to Amazon S3 is no longer available: please migrate to the s3a:// client
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at water.persist.PersistHdfs.exists(PersistHdfs.java:461)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at water.persist.PersistManager.exists(PersistManager.java:439)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at water.fvec.Frame.export(Frame.java:1433)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at water.api.FramesHandler.export(FramesHandler.java:247)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at java.lang.reflect.Method.invoke(Method.java:498)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at water.api.Handler.handle(Handler.java:63)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at water.api.RequestServer.serve(RequestServer.java:482)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at water.api.RequestServer.doGeneric(RequestServer.java:327)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at water.api.RequestServer.doPost(RequestServer.java:253)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:848)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:584)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1180)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:512)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1112)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:119)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at water.webserver.jetty9.Jetty9ServerAdapter$LoginHandler.handle(Jetty9ServerAdapter.java:119)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:119)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.server.Server.handle(Server.java:534)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:320)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:283)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:108)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:303)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 	at java.lang.Thread.run(Thread.java:748)
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: Caught exception: 
12-07 12:22:39.985 10.20.30.189:54445    2256   #44334-31 ERRR: 

```

**Comment by Pscheidl** [comment]
This is also an error to solve:

```plain
12-07 12:22:28.446 10.20.30.189:54445    2256   #44334-34 INFO: GET /3/Frames/AirlinesTest.hex, parms: {column_offset=0, full_column_count=-1, row_count=10, row_offset=0, column_count=-1}
12-07 12:22:28.455 10.20.30.189:54445    2256   #44334-52 INFO: POST /99/ImportSQLTable, parms: {select_query=select * from airlinestest, password=, username=hive, connection_url=jdbc:hive2://localhost:10000/default}
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: java.lang.RuntimeException: SQLException: Error while processing statement: FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: Failed to connect and read from SQL database with connection_url: jdbc:hive2://localhost:10000/default
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at water.jdbc.SQLManager.importSqlTable(SQLManager.java:150)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at water.api.ImportSQLTableHandler.importSQLTable(ImportSQLTableHandler.java:25)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at java.lang.reflect.Method.invoke(Method.java:498)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at water.api.Handler.handle(Handler.java:63)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at water.api.RequestServer.serve(RequestServer.java:482)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at water.api.RequestServer.doGeneric(RequestServer.java:327)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at water.api.RequestServer.doPost(RequestServer.java:253)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:848)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:584)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1180)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:512)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1112)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:119)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at water.webserver.jetty9.Jetty9ServerAdapter$LoginHandler.handle(Jetty9ServerAdapter.java:119)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:119)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.server.Server.handle(Server.java:534)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:320)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:283)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:108)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:303)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at java.lang.Thread.run(Thread.java:748)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: Caused by: java.sql.SQLException: Error while processing statement: FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.apache.hive.jdbc.HiveStatement.waitForOperationToComplete(HiveStatement.java:359)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.apache.hive.jdbc.HiveStatement.execute(HiveStatement.java:260)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at org.apache.hive.jdbc.HiveStatement.executeUpdate(HiveStatement.java:465)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	at water.jdbc.SQLManager.importSqlTable(SQLManager.java:74)
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 	... 34 more
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: Caught exception: 
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: ERROR MESSAGE:
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: SQLException: Error while processing statement: FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: Failed to connect and read from SQL database with connection_url: jdbc:hive2://localhost:10000/default
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: 
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: ; Stacktrace: [java.lang.RuntimeException: SQLException: Error while processing statement: FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask
12-07 12:22:37.348 10.20.30.189:54445    2256   #44334-52 ERRR: Failed to connect and read from SQL database with connection_url: jdbc:hive2://localhost:10000/default,     water.jdbc.SQLManager.importSqlTable(SQLManager.java:150),     water.api.ImportSQLTableHandler.importSQLTable(ImportSQLTableHandler.java:25),     sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method),     sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62),     sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43),     java.lang.reflect.Method.invoke(Method.java:498),     water.api.Handler.handle(Handler.java:63),     water.api.RequestServer.serve(RequestServer.java:482),     water.api.RequestServer.doGeneric(RequestServer.java:327),     water.api.RequestServer.doPost(RequestServer.java:253),     javax.servlet.http.HttpServlet.service(HttpServlet.java:707),     javax.servlet.http.HttpServlet.service(HttpServlet.java:790),     org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:848),     org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:584),     org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1180),     org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:512),     org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1112),     org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141),     org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:119),     org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134),     water.webserver.jetty9.Jetty9ServerAdapter$LoginHandler.handle(Jetty9ServerAdapter.java:119),     org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:119),     org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134),     org.eclipse.jetty.server.Server.handle(Server.java:534),     org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:320),     org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251),     org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:283),     org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:108),     org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93),     org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:303),     org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148),     org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136),     org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671),     org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589),     java.lang.Thread.run(Thread.java:748), Caused by:java.sql.SQLException: Error while processing statement: FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask,     org.apache.hive.jdbc.HiveStatement.waitForOperationToComplete(HiveStatement.java:359),     org.apache.hive.jdbc.HiveStatement.execute(HiveStatement.java:260),     org.apache.hive.jdbc.HiveStatement.executeUpdate(HiveStatement.java:465),     water.jdbc.SQLManager.importSqlTable(SQLManager.java:74),     water.api.ImportSQLTableHandler.importSQLTable(ImportSQLTableHandler.java:25),     sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method),     sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62),     sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43),     java.lang.reflect.Method.invoke(Method.java:498),     water.api.Handler.handle(Handler.java:63),     water.api.RequestServer.serve(RequestServer.java:482),     water.api.RequestServer.doGeneric(RequestServer.java:327),     water.api.RequestServer.doPost(RequestServer.java:253),     javax.servlet.http.HttpServlet.service(HttpServlet.java:707),     javax.servlet.http.HttpServlet.service(HttpServlet.java:790),     org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:848),     org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:584),     org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1180),     org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:512),     org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1112),     org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141),     org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:119),     org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134),     water.webserver.jetty9.Jetty9ServerAdapter$LoginHandler.handle(Jetty9ServerAdapter.java:119),     org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:119),     org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134),     org.eclipse.jetty.server.Server.handle(Server.java:534),     org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:320),     org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251),     org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:283),     org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:108),     org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93),     org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:303),     org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148),     org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136),     org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671),     org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589),     java.lang.Thread.run(Thread.java:748)];parms={select_query=select * from airlinestest, password=, username=hive, connection_url=jdbc:hive2://localhost:10000/default}
12-07 12:22:37.901 10.20.30.189:54445    2256   #44334-38 INFO: POST /3/LogAndEcho, parms: {message=------------------------------------------------------------}
```

**Comment by michal-raska** [comment]
Note to myself, I need to resolve comments regarding `testCompile` deps.

**Comment by Pscheidl** [comment]
Ok, what was done so far:

1. Rebased on latest master
1. Changed base of this PR to master
1. Splitted hadoop builds into two modules. `h2o-hadoop-2` and `h2o-hadoop-3`. The naming convention is obvious. Currently, only CDH6 is assembled in `h2o-hadoop-3`
1. Adjusted `h2o-mapreduce-3-generic` to Hadoop 3. Only different dependencies were required, otherwise works flawlessly.
1. Resolved issue with Amazon S3 test. `S3N` protocol is deprecated in new HDP and only `s3a` should be used from the set of two we use, so I kept S3a. Fixed.
1. The test suits for `hadoop-2` and `hadoop-3` became disjoint, so each module has its own test suit. Right now, those are almost exactly the same (see previous point), but in future I assume there will be need to test hadoop-2 and hadoop-3 independently.
1. Resolved mapreduce-generic module dependencies stacktrace mentioned above. This is fixed.

What needs to be discussed:

### Hive-on-MR configuration
There is an issue with configuration of HIVE2 in our docker that needs to be discussed. In previous versions, there is already a warning message :

```WARNING: Hive-on-MR is deprecated in Hive 2 and may not be available in the future versions. Consider using a different execution engine (i.e. spark, tez) or using Hive 1.X releases.
No rows affected (18.541 seconds)
```

On newer versions that are used when CDH-6 tests are executed, this warning is gone and it just fails. They removed the supported for Hive-on-MR. Related issue is here: https://issues.apache.org/jira/browse/HIVE-12300

This makes the HIVE2 tests fail. Reason is simple. Our chosen strategy is to extract the results of a query into separate temporary table and then let the nodes do something with the data. The resulting query executed by JDBC looks like that: `CREATE TABLE tmp AS SELECT * FROM airlinestest`. This makes HIVE2 trigger a new MR job and fail. A simple select does work - probably because a MR job is not triggers (cache used ?).

So far, we believe the correct solution is to adjust the Docker to use Tez or Spark as suggested. I know nothing about Tez.

The root cause of this is quite hard to debug. We could agree on doing some "heuristics" in SQLManager: If the driver is HIVE2 and a select query with the above-mentioned strategy fails, we could do the following:
- Use plain select as a fallback (just thinking out loud)
- Warn the user to check their environment. We are able to detect this issue by the response we receive via JDBC driver. But it does not appear in the Stacktrace sadly.

### Docker adjustments
Related to previous point. Me & @michal-raska need to re-configure or create separate Dockerfile with proper HIVE configuration in order for the HDP3-related tests to work. Not time consuming, but we need to double-check these changes done are what we want before I drag him into it.

File: h2o-parsers/h2o-parquet-compat/h2o-parquet-v17-compat/build.gradle (hunk 3 above)
Comment (michalkurka) [reply h2o-parsers/h2o-parquet-compat/h2o-parquet-v17-compat/build.gradle]: Can we get around this? It is certainly not convenient that every single module needs to declare a testCompile dependency. I would almost say that is not correct :)

**Comment by michalkurka** [comment]
@Pscheidl why were the python tests moved? I would expect them to be hadoop version independent. Is it because of S3N?

**Comment by Pscheidl** [comment]
@michalkurka Yes, that reason only right now.

**Comment by Pscheidl** [comment]
Rebased on top of latest master.

**Comment by Pscheidl** [comment]
### Summary
- HIVE test has been moved to Hadoop 2 folder, as our docker container has misconfigured HIVE2 environment (to be fixed ASAP in future).
- Minor build tweaks - removing unused dependencies, properties & cleanup
- Rebased on top of lastest master to make sure all the previous jetty-related changes are in place.

**Comment by Pscheidl** [comment]
Rebased on latest master once again. This includes Jetty 8 update from master. 

Added Guava 20 with relocations in latest commit to honor the work of @jakubhava in our latest module layout.
